march_hardware/src/IMotionCube.cpp

```diff
@@ -84,9 +84,19 @@ void IMotionCube::mapMosiPDOs()
 }
 
 // Set configuration parameters to the IMC
-void IMotionCube::writeInitialSettings(uint8_t cycle_time)
+bool IMotionCube::writeInitialSettings(uint8_t cycle_time)
 {
-  ROS_DEBUG("IMotionCube::writeInitialSettings");
+  int checksum_verified = verifySetup();
+  ROS_INFO("The .sw file for slave %d is %s", this->slaveIndex,
+           checksum_verified ? "equal to the setup of the drive." :
+                               "not equal to the setup of the drive, downloading is necessary");
```

Comment:
I think them not being equal should be considered a special event. Therefore I suggest make it a debug when successful and a warning when unsuccessful. It should not be necessary in normal use.

Reply:
This is a good suggestion, I will implement this shortly.